packages/wp-story-editor/src/components/fontCheck/index.js

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright 2022 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+/**
+ * External dependencies
+ */
+import { useEffect, useCallback, useState } from '@googleforcreators/react';
+import {
+  useStory,
+  useConfig,
+  useAPI,
+  getInUseFontsForPages,
+} from '@googleforcreators/story-editor';
+
+/**
+ * Internal dependencies
+ */
+import { useFeature } from 'flagged';
+import { FontCheckDialog } from './fontCheckDialog';
+
+export const FontCheck = () => {
+  const { dashboardLink } = useConfig();
+  const notifyDeletedFonts = useFeature('notifyDeletedFonts');
+  const storyPages = useStory(({ state: { pages } }) => pages);
+
+  const {
+    actions: { getFonts },
+  } = useAPI();
+
+  const [checkedFonts, hasCheckedFonts] = useState(false);
+  const [missingFont, setMissingFont] = useState(null);
+  const [showDialog, setShowDialog] = useState(false);
+  const closeDialog = useCallback(
+    (evt) => {
+      if (evt) {
+        // redirect when clicking outside the dialog
+        window.location = dashboardLink;
+      }
+      setShowDialog(false);
+    },
+    [dashboardLink]
+  );
```

Comment:
To make this less confusing, I suggest passing 2 separate props to `FontCheckDialog` instead of using this callback for 2 things.

Reply:
There's now 2 functions